pkg/kubectl/cmd/config/get_context.go

```diff
@@ -0,0 +1,169 @@
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	"k8s.io/kubernetes/pkg/kubectl"
+
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+type GetContextOptions struct {
+	configAccess clientcmd.ConfigAccess
+	contextName  string
+}
+
+const (
+	get_context_long = `Displays one or many contexts from the kubeconfig file.`
+
+	get_context_example = `
+  $ kubectl config get-context
+    NAME                 CLUSTER         AUTHINFO     NAMESPACE
+    federal-context      horse-cluster   green-user   chisel-ns
+    queen-anne-context   big-cluster     blue-user    saw-ns
+
+  $ kubectl config get-context --no-headers
+    federal-context      horse-cluster   green-user   chisel-ns
+    queen-anne-context   big-cluster     blue-user    saw-ns
+
+  $ kubectl config get-context -o name
+    NAME
+    queen-anne-context
+    federal-context
+
+  $ kubectl config get-context -o name --no-headers
+    federal-context
+    queen-anne-context`
+)
+
+// NewCmdGetContext creates a command object for the "get-context" action, which
+// retrieves one or more contexts from a kubeconfig.
+func NewCmdConfigGetContext(out io.Writer, configAccess clientcmd.ConfigAccess) *cobra.Command {
+	options := &GetContextOptions{configAccess: configAccess}
+
+	cmd := &cobra.Command{
+		Use:     "get-context [(-o|--output=)detail|name)]",
+		Short:   "Get one or many contexts",
+		Long:    get_context_long,
+		Example: get_context_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(cmd, args))
+			cmdutil.CheckErr(options.RunGetContext(cmd, out))
+		},
+	}
+	cmd.Flags().StringP("output", "o", "", "Output format. One of: detail|name.")
+	cmd.Flags().Bool("no-headers", false, "Don't print column headers.")
+
+	return cmd
+}
+
+func (o *GetContextOptions) Complete(cmd *cobra.Command, argsIn []string) error {
+	if len(argsIn) == 1 {
+		o.contextName = argsIn[0]
+	}
+	defaultOutputFormat := "detail"
+	outputFormat := cmdutil.GetFlagString(cmd, "output")
+	if outputFormat == "" {
+		cmd.Flags().Set("output", defaultOutputFormat)
+	}
+
+	return nil
+}
+
+func (o GetContextOptions) Validate(config *clientcmdapi.Config, outputFormat string) error {
+	if outputFormat != "detail" && outputFormat != "name" {
+		return fmt.Errorf("output must be one of 'detail' or 'name': %q.", outputFormat)
+	}
+
+	if len(o.contextName) == 0 {
+		// context name is optional
+		return nil
+	}
+
+	for name := range config.Contexts {
+		if name == o.contextName {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("no context exists with the name: %q.", o.contextName)
+}
+
+// RunGetContext implements all the necessary functionality for context retrieval.
+func (o GetContextOptions) RunGetContext(cmd *cobra.Command, output io.Writer) error {
```

Comment:
> avoid passing the cmd and the output here. Those can be values on your options struct.

We've done this openshift by passing a printer function that we build in `Complete`.  It keeps the code down in run cleaner.